MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -2918,5 +2918,27 @@ public virtual long GetRunTimeTicksForPlayState()
         public static readonly IReadOnlyCollection<ExtraType> DisplayExtraTypes = new[] { Model.Entities.ExtraType.BehindTheScenes, Model.Entities.ExtraType.Clip, Model.Entities.ExtraType.DeletedScene, Model.Entities.ExtraType.Interview, Model.Entities.ExtraType.Sample, Model.Entities.ExtraType.Scene };
 
         public virtual bool SupportsExternalTransfer => false;
+
+        public override bool Equals(object obj)
+        {
+            Logger.LogDebug("Comparing me ({0}) to generic them ({1})", this, obj);
+            return this.Equals(obj as BaseItem);
+        }
+
+        public bool Equals(BaseItem item)
+        {
+            Logger.LogDebug("Comparing me ({0}) to specific them ({1})", this, item);
+            if (item == null)
+            {
+                return false;
+            }
+
+            return Id == item.Id;
```

Comment:
Need to check for null `Id` here. `object.Equals(Id, item?.Id)` should do the trick I think (and then you can get rid of the null check above as well)

Reply:
neat, thx :)